aioesphomeapi/model.py

```diff
@@ -757,6 +756,28 @@ class AlarmControlPanelEntityState(EntityState):
     )
 
 
+# ==================== TEXT ====================
+class TextMode(APIIntEnum):
+    TEXT = 0
+    PASSWORD = 1
+
+
+@dataclass(frozen=True)
+class TextInfo(EntityInfo):
+    min_length: int = 0
+    max_length: int = 255
+    pattern: str = ""
+    mode: Optional[TextMode] = converter_field(
+        default=TextMode.TEXT, converter=TextMode.convert
+    )
+
+
+@dataclass(frozen=True)
```

Comment:
We could use the _frozen_dataclass_decorator here so it uses less ram on newer python